src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            machines.addAll(new BulldozerProducer().get());
+        } else if (type.equals(Excavator.class)) {
+            machines.addAll(new ExcavatorProducer().get());
+        } else if (type.equals(Truck.class)) {
+            machines.addAll(new TruckProducer().get());
+        }
```

Comment:
these producers are created each time the method is called. Could you somehow improve your solution here?